account_payment_return/models/payment_return.py

```diff
@@ -108,6 +108,13 @@ def _prepare_invoice_returned_cancel_vals(self):
         return {'returned_payment': False}
 
     @api.multi
+    def unlink(self):
+        if self.filtered(lambda x: x.state == 'done'):
+            raise UserError(_(
+                "You can not remove a payment return if state is 'Done'"))
+        return super(PaymentReturn, self).unlink()
```

Comment:
I'd implement this as a [global ir.rule](https://www.odoo.com/documentation/8.0/reference/security.html#record-rules) instead.